README.md
# StartPi


StartPi est un projet originalement créé par [@zonguin](https://steemit.com/@zonguin)

Phase de développement : ALPHA

------

StartPi est un script qui permet de configurer, de sécuriser et de booster votre Raspberry Pi en quelques clics.

### Il permet pour le moment de :
- Mettre à jour le système
- Changer le nom de la raspberry (hostname)
- Changer le nom de l' utilisateur Pi
- Changer le port SSH
- Securiser le SSH
- Mettre à jour nodejs
- Installer ruby
- Installer les prérequis pour installer des bots
- Installer certains utilitaires :
    - usbmount
- Supprimer automatiquements certains paquets inutiles (paquets graphiques)



#### Pour le moment, voici les languages supportés :
- *fr_FR*
- *en_EN*

------

### Installation :


<code>
    curl -sSL https://raw.githubusercontent.com/ZongUin/StartPi/master/startpi.sh | bash
</code>

------

StartPi is a script that allows you to configure, secure and boost your Raspberry Pi in a few clicks.

### It allows for the moment to :
- Update the system
- Change  RPi HostName
- Change user name Pi
- Change SSH port
- Secure SSH
- Update nodejs
- Install ruby
- Install the prerequisites to install bots
- Install some Utilities :
    - usbmount
- Remove automaticaly some useless packets (graphical packets)

#### For the moment, here are the supported languages :
- *en_FR*
- *en_EN*


changelog.md
# Changelog

## V Alpha-0.2

    #### Language support :

    Full English support !

    #### New fonctionnalities :
        - Added SSHSecurity : Auto secure the SSH connection
        - Added NodeUpdate : Update nodejs automaticaly
        - Added Ruby : Auto install Ruby
        - Added OtherUpdates : Install some tools for other softwares that might be installed by StartPi
        - Added DelGraphic : Remove automaticaly some useless packets (graphical packets)
        - Added usbmount : Automatically install usbmount

## V Alpha-0.21

- More English Support
- English version commented
- Added .gitignore file



en.sh
################################################################
##                       English Version                      ##
################################################################

RED="\033[0;31m" #set colors for terminal
GREEN="\033[0;32m"
YELLOW="\033[0;33m"
RESET="\033[0m"

echo "Welcome into StartPi configuration tool for Raspberry Pi by @zonguin !"

{ if
(whiptail --title "RPiUpdate" --yes-button "No" --no-button "Yes" --yesno "Update system (Extremely recommended) ?" 10 80) # Menu
then
echo "$CURRENTUSER not activate RPiUpdate" | sudo tee --append /etc/startpi/log.txt # logging
else
echo "$CURRENTUSER activate RPiUpdate" | sudo tee --append /etc/startpi/log.txt

echo "Updating system..."

sudo apt-get install -y
sudo apt-get upgrade -y #install and upgrade system
sudo apt-get dist-upgrade

echo "Update finished !"


fi }

{ if
(whiptail --title "ChangeName" --yes-button "No" --no-button "Yes" --yesno "Change user pi name ?  - You must be connected as root (command : su) ?" 10 80) # Menu
then
echo "$CURRENTUSER not activate ChangeName" | sudo tee --append /etc/startpi/log.txt #logging
else
echo "$CURRENTUSER activate ChangeName" | sudo tee --append /etc/startpi/log.txt

echo "Changing pi name..."

NEW_PINAME=$(whiptail --inputbox "Entrez le nouveau nom d' utilisateur" 10 80 "pi" 3>&1 1>&2 2>&3)
sudo usermod --login $NEW_PINAME  --home /home/$NEW_PINAME --move-home pi #change pi user name and move files to the new folder
echo "Done !"


fi }

{ if
(whiptail --title "ChangeSSHport" --yes-button "No" --no-button "Yes" --yesno "Change ssh port (Recommended for more security) ?" 10 80)
then
echo "$CURRENTUSER not activate ChangeSSHport" | sudo tee --append /etc/startpi/log.txt
else
echo "$CURRENTUSER activate ChangeSSHport" | sudo tee --append /etc/startpi/log.txt

echo "Updating system..."

SSH_PORT=$(whiptail --inputbox "Enter the new port (choose beetween 1024 and 65535) " 10 80 "22" 3>&1 1>&2 2>&3) #choose a correct port
    { if (("$SSH_PORT" > 220 && "$SSH_PORT" < 65535)) # check that it is a correct port
    then
    sed -i "s/^Port .*/Port "$SSH_PORT"/g" /etc/ssh/sshd_config #change the port in sshd_config
    echo "Done !"
    else
    echo "$SSH_PORT" # incorrect port --> error
    echo "Incorrect port"
    fi }

fi }

{ if
(whiptail --title "SSHSecurity" --yes-button "No" --no-button "Yes" --yesno "Apply security modifications to the SSH ?" 10 80) # Menu
then
echo "$CURRENTUSER not activate SSHSecurity" | sudo tee --append /etc/startpi/log.txt
else
echo "$CURRENTUSER activate SSHSecurity" | sudo tee --append /etc/startpi/log.txt

echo "Loading..."

SSH_TRY=$(whiptail --inputbox "How many Auth tries before SSH auto-logout ? " 10 80 "2" 3>&1 1>&2 2>&3) #ask user

sed -i "$ a MaxAuthTries "$SSH_TRY"" /etc/ssh/sshd_config #add security line : MaxAuthTries

SSH_USERS=$(whiptail --inputbox "Which user is allowed to use the SSH [only one user] " 10 80 "pi" 3>&1 1>&2 2>&3)

sed -i "$ a AllowUsers "$SSH_USERS"" /etc/ssh/sshd_config # add secrity line : AllowUsers

sed -i "$ a LoginGraceTime 1m" /etc/ssh/sshd_config


echo "SSH Secured !"

fi }


{ if
(whiptail --title "ChangeHostName" --yes-button "No" --no-button "Yes" --yesno "Change Raspberry Name ?" 10 80)
then
echo "$CURRENTUSER not activate ChangeHostName" | sudo tee --append /etc/startpi/log.txt
else
echo "$CURRENTUSER activate ChangeHostName" | sudo tee --append /etc/startpi/log.txt

echo "Loading..."

OLD_HOST=`cat /etc/hostname | tr -d " \t\n\r"` # check old hostname
NEW_HOST=$(whiptail --inputbox "Please enter a hostname" 10 80 "$OLD_HOSTNAME" 3>&1 1>&2 2>&3)
sudo cp /etc/hosts /etc/startpi/backups/host/ # backup the hosts file
sudo cp /etc/hostname /etc/startpi/backups/host/
sudo echo "NEW_HOST=$NEW_HOST"
sudo echo "$NEW_HOST" > /etc/hostname #change hostname
sudo sed -i "s/127.0.1.1.*$OLD_HOST/127.0.1.1\t$NEW_HOST/g" /etc/hosts #remplace the host name

echo "Done !"

fi }


{ if
(whiptail --title "NodeUpdate" --yes-button "No" --no-button "Yes" --yesno "Update nodejs (Recommended to create some bots) ?" 10 80)
then
echo "$CURRENTUSER not activate NodeUpdate" | sudo tee --append /etc/startpi/log.txt
else
echo "$CURRENTUSER activate NodeUpdate" | sudo tee --append /etc/startpi/log.txt

echo "Updating nodejs..."

curl -sL https://deb.nodesource.com/setup_10.x | sudo -E bash - # install the installation script
sudo apt install nodejs


echo "Updated !"


fi }

{ if
(whiptail --title "Ruby" --yes-button "No" --no-button "Yes" --yesno "Installer Ruby (Recommended to create some bots) ?" 10 80)
then
echo "$CURRENTUSER not activate Ruby" | sudo tee --append /etc/startpi/log.txt
else
echo "$CURRENTUSER activate Ruby" | sudo tee --append /etc/startpi/log.txt

echo "Installation..."

apt-get install dirmngr
gpg --keyserver hkp://keys.gnupg.net --recv-keys 409B6B1796C275462A1703113804BB82D39DC0E3 7D2BAF1CF37B13E2069D6956105BD0E739499BDB # add keys for the next command
curl -sSL https://get.rvm.io | bash -s stable -s stable --autolibs=3 --ruby --rails # install ruby
sudo /usr/sbin/usermod -a -G rvm $USER # add permissions
sudo chown root.rvm /etc/profile.d/rvm.sh
source /usr/local/rvm/scripts/rvm # launch ruby


echo "Installation done !"


fi }

{ if
(whiptail --title "OtherUpdates" --yes-button "No" --no-button "Yes" --yesno "Update / Install some others frameworks [recommended for some Auto Install in StartPi] ?" 10 80)
then
echo "$CURRENTUSER not activate OtherUpdates" | sudo tee --append /etc/startpi/log.txt
else
echo "$CURRENTUSER activate OtherUpdates" | sudo tee --append /etc/startpi/log.txt

echo "Updating..."


mkdir  /tmp/startpi
cd /tmp/startpi
sudo apt-get update
wget https://cpan.metacpan.org/authors/id/A/AN/ANDK/CPAN-2.16.tar.gz # download cpan update
tar zxf CPAN-2.16.tar.gz
perl Makefile.PL
make test
sudo make install # install cpan
sudo apt-get install yarn
npm install -g steem discord.js moment bufferutil erlpack node-opus opusscript  sodium libsodium-wrappers uws mathjs convnetjs # install npm modules
sudo apt-get -y install git
apt-get install pm2

echo "Updated !"

rm -rf /tmp/startpi


fi }

{ if
(whiptail --title "DelGraphic" --yes-button "No" --no-button "Yes" --yesno "Delete some useless packets [Only raspbian Lite / No graphic interface] ?" 10 80)
then
echo "$CURRENTUSER not activate DelGraphic" | sudo tee --append /etc/startpi/log.txt
else
echo "$CURRENTUSER activate DelGraphic" | sudo tee --append /etc/startpi/log.txt

echo "Cleaning..."

apt-get -y remove --purge xserver-common x11-common gnome-icon-theme gnome-themes-standard penguinspuzzle # remove packets
apt-get -y remove --purge desktop-base desktop-file-utils hicolor-icon-theme raspberrypi-artwork omxplayer
apt-get -y autoremove

echo "Done !"


fi }

whiptail --title "partie Utilitaires" --yes-button "No" --no-button "yes" --yesno "you enter in part [Utilities Installation] ?" 10 80
{ if
(whiptail --title "usbmount" --yes-button "No" --no-button "Yes" --yesno "Install usbmount to mount / unmount usb devices automaticaly when plugged ?" 10 80)
then
echo "$CURRENTUSER not activate usbmount" | sudo tee --append /etc/startpi/log.txt
else
echo "$CURRENTUSER activate usbmount" | sudo tee --append /etc/startpi/log.txt

echo "Installation..."

apt-get install usbmount

echo "Installation done !"


fi }

echo "Program ended. Thanks for using StartPi by @zonguin !"



################################################################
##                   Stopped English Version                  ##
################################################################


startpi.sh

#!/bin/sh
################################################################
##                     StartPi by @zonguin                    ##
################################################################
##           Please start the script with root user           ##
################################################################

# Set Colors for the echo command

RED="\033[0;31m"
GREEN="\033[0;32m"
YELLOW="\033[0;33m"
RESET="\033[0m"


{ if
which whiptail >/dev/null;
thenre
:
else
sudo apt-get install -y whiptail # install whiptail to create menus
fi }

sudo -l > /dev/null # Verify the user has permissions

CURRENTUSER="$(whoami)"
# This get the username of the user

mkdir /etc/startpi # create startpi folders
mkdir /etc/startpi/backups
mkdir /etc/startpi/backups/host
cp /etc/startpi/log.txt /etc/startpi/logback.txt # save last backup : 2 backups will be available
rm /etc/startpi/log.txt  # start a new backup by deleting previous backup

chmod +x en.sh # Give permissions to execute StartPi
chmod +x fr.sh




{ if
(whiptail --title "Langage" --yes-button "No" --no-button "Yes" --yesno "Use French Version ? " 10 80)
then
echo "$CURRENTUSER use English langage" | sudo tee --append /etc/startpi/log.txt # Add data to the log file
echo "Starting with English Version..."

sudo ./en.sh

else
echo "$CURRENTUSER a utilisé le language Français" | sudo tee --append /etc/startpi/log.txt

echo "Starting with French Version..."

echo "Bienvenue dans l' assistant de configuration StartPi par @zonguin !"

sudo ./fr.sh

fi }

sudo systemctl restart ssh

## END
unset CURRENTUSER
unset NEW_PINAME
unset SSH_PORT
unset SSH_TRY
unset SSH_USERS


fr.sh
################################################################
##                       French Version.                      ##
################################################################

RED="\033[0;31m"
GREEN="\033[0;32m"
YELLOW="\033[0;33m"
RESET="\033[0m"

{ if
(whiptail --title "RPiUpdate" --yes-button "Non" --no-button "Oui" --yesno "Mettre à jour le système (Vivement recommandé) ?" 10 80)
then
echo "${YELLOW}>>>$CURRENTUSER n' a pas activé RPiUpdate${RESET}\n" | sudo tee --append /etc/startpi/log.txt
else
echo -e "${YELLOW}>>> $CURRENTUSER a activé RPiUpdate${RESET}\n" | sudo tee --append /etc/startpi/log.txt

echo -e "${GREEN}>>> Mise à jour du système...${RESET}\n"

sudo apt-get install -y
sudo apt-get upgrade -y
sudo apt-get dist-upgrade

echo "${GREEN}>>> Mise à jour effectuée !${RESET}\n"


fi }

{ if
(whiptail --title "ChangeName" --yes-button "Non" --no-button "Oui" --yesno "Changer le nom de l' utilisateur pi (recommandé pour plus de sécurité) - Il faut être connecté avec l' utilisateur root (commande : su) ?" 10 80)
then
echo "$CURRENTUSER n' a pas activé ChangeName" | sudo tee --append /etc/startpi/log.txt
else
echo "$CURRENTUSER a activé ChangeName" | sudo tee --append /etc/startpi/log.txt

echo "Changement du nom..."

NEW_PINAME=$(whiptail --inputbox "Entrez le nouveau nom d' utilisateur" 10 80 "pi" 3>&1 1>&2 2>&3)
sudo usermod --login $NEW_PINAME  --home /home/$NEW_PINAME --move-home pi
echo "Changement effectué !"


fi }

{ if
(whiptail --title "ChangeSSHport" --yes-button "Non" --no-button "Oui" --yesno "Changer le port ssh (Recommandé pour plus de sécurité) ?" 10 80)
then
echo "$CURRENTUSER n' a pas activé ChangeSSHport" | sudo tee --append /etc/startpi/log.txt
else
echo "$CURRENTUSER a activé ChangeSSHport" | sudo tee --append /etc/startpi/log.txt

echo "Mise à jour du système..."

SSH_PORT=$(whiptail --inputbox "Entrez le nouveau port (choisir entre 1024 et 65535) " 10 80 "22" 3>&1 1>&2 2>&3)
    { if (("$SSH_PORT" > 220 && "$SSH_PORT" < 65535))
    then
    cp /etc/ssh/sshd_config /etc/startpi/backups/
    sed -i "s/^Port .*/Port "$SSH_PORT"/g" /etc/ssh/sshd_config
    echo "Mise à jour effectuée !"
    else
    echo "$SSH_PORT"
    echo "Erreur : Port incorrect"
    fi }

fi }

{ if
(whiptail --title "SSHSecurity" --yes-button "Non" --no-button "Oui" --yesno "Appliquer des améliorations de sécurité SSH ?" 10 80)
then
echo "$CURRENTUSER n' a pas activé SSHSecurity" | sudo tee --append /etc/startpi/log.txt
else
echo "$CURRENTUSER a activé SSHSecurity" | sudo tee --append /etc/startpi/log.txt

echo "Chargement..."

SSH_TRY=$(whiptail --inputbox "Combien de tentatives de connexion autoriser ? " 10 80 "2" 3>&1 1>&2 2>&3)

sed -i "$ a MaxAuthTries "$SSH_TRY"" /etc/ssh/sshd_config

SSH_USERS=$(whiptail --inputbox "Quels utilisateurs autorisés ? [Un seul uniquement] " 10 80 "pi" 3>&1 1>&2 2>&3)

sed -i "$ a AllowUsers "$SSH_USERS"" /etc/ssh/sshd_config

sed -i "$ a LoginGraceTime 1m" /etc/ssh/sshd_config


echo "SSH Sécurisé !"

fi }


{ if
(whiptail --title "ChangeHostName" --yes-button "Non" --no-button "Oui" --yesno "Changer le nom de la raspberry ?" 10 80)
then
echo "$CURRENTUSER n' a pas activé ChangeHostName" | sudo tee --append /etc/startpi/log.txt
else
echo "$CURRENTUSER a activé ChangeHostName" | sudo tee --append /etc/startpi/log.txt

echo "Changement..."

OLD_HOST=`cat /etc/hostname | tr -d " \t\n\r"`
NEW_HOST=$(whiptail --inputbox "Entrez un hostname (le nouveau nom)" 10 80 "$OLD_HOSTNAME" 3>&1 1>&2 2>&3)
sudo cp /etc/hosts /etc/startpi/backups/host/
sudo cp /etc/hostname /etc/startpi/backups/host/
sudo echo "NEW_HOST=$NEW_HOST"
sudo echo "$NEW_HOST" > /etc/hostname
sudo sed -i "s/127.0.1.1.*$OLD_HOST/127.0.1.1\t$NEW_HOST/g" /etc/hosts

echo "Changement effectué !"

fi }

echo "Programme terminé. Merci d' avoir utilisé StartPi par @zonguin !"



{ if
(whiptail --title "NodeUpdate" --yes-button "Non" --no-button "Oui" --yesno "Mettre à jour nodejs (Recommandé pour de nombreux bots) ?" 10 80)
then
echo "$CURRENTUSER n' a pas activé NodeUpdate" | sudo tee --append /etc/startpi/log.txt
else
echo "$CURRENTUSER a activé NodeUpdate" | sudo tee --append /etc/startpi/log.txt

echo "Mise à jour de nodejs..."

curl -sL https://deb.nodesource.com/setup_10.x | sudo -E bash -
sudo apt install nodejs


echo "Mise à jour effectuée !"


fi }

{ if
(whiptail --title "Ruby" --yes-button "Non" --no-button "Oui" --yesno "Installer Ruby (Recommandé pour de nombreux bots) ?" 10 80)
then
echo "$CURRENTUSER n' a pas activé Ruby" | sudo tee --append /etc/startpi/log.txt
else
echo "$CURRENTUSER a activé Ruby" | sudo tee --append /etc/startpi/log.txt

echo "Installation..."

apt-get install dirmngr
gpg --keyserver hkp://keys.gnupg.net --recv-keys 409B6B1796C275462A1703113804BB82D39DC0E3 7D2BAF1CF37B13E2069D6956105BD0E739499BDB
curl -sSL https://get.rvm.io | bash -s stable -s stable --autolibs=3 --ruby --rails
sudo /usr/sbin/usermod -a -G rvm $USER
sudo chown root.rvm /etc/profile.d/rvm.sh
source /usr/local/rvm/scripts/rvm


echo "Installation effectuée !"


fi }

{ if
(whiptail --title "OtherUpdates" --yes-button "Non" --no-button "Oui" --yesno "Mettre à jour / installer d' autres frameworks (Utiles pour les installateurs suivants) ?" 10 80)
then
echo "$CURRENTUSER n' a pas activé OtherUpdates" | sudo tee --append /etc/startpi/log.txt
else
echo "$CURRENTUSER a activé OtherUpdates" | sudo tee --append /etc/startpi/log.txt

echo "Mise à jour..."


mkdir  /tmp/startpi
cd /tmp/startpi
sudo apt-get update
wget https://cpan.metacpan.org/authors/id/A/AN/ANDK/CPAN-2.16.tar.gz
tar zxf CPAN-2.16.tar.gz
cd CPAN-2.16
perl Makefile.PL
make test
sudo make install
sudo apt-get install yarn
npm install -g steem discord.js moment bufferutil erlpack node-opus opusscript  sodium libsodium-wrappers uws mathjs convnetjs
sudo apt-get -y install git
apt-get install dirmngr

echo "Mise à jour effectuée !"

rm -rf /tmp/startpi


fi }

{ if
(whiptail --title "DelGraphic" --yes-button "Non" --no-button "Oui" --yesno "Supprimer certains paquets inutiles [Version raspbian Lite uniquement / sans interface graphique] ?" 10 80)
then
echo "$CURRENTUSER n' a pas activé DelGraphic" | sudo tee --append /etc/startpi/log.txt
apt-get -y autoremove
else
echo "$CURRENTUSER a activé DelGraphic" | sudo tee --append /etc/startpi/log.txt

echo "Nettoyage en cours..."

apt-get -y remove --purge xserver-common x11-common gnome-icon-theme gnome-themes-standard penguinspuzzle
apt-get -y remove --purge desktop-base desktop-file-utils hicolor-icon-theme raspberrypi-artwork omxplayer
apt-get -y autoremove

echo "nettoyage effectué !"


fi }

whiptail --msgbox "Let's change pihole password" 10 80 1

{ if
(whiptail --title "usbmount" --yes-button "Non" --no-button "Oui" --yesno "Installer usbmount pour monter/démonter automatiquement les périphériques USB au branchement ?" 10 80)
then
echo "$CURRENTUSER n' a pas activé usbmount" | sudo tee --append /etc/startpi/log.txt
else
echo "$CURRENTUSER a activé usbmount" | sudo tee --append /etc/startpi/log.txt

echo "Installation..."

apt-get install usbmount

echo "Installation effectuée !"


fi }

whiptail --msgbox "Vous allez entrer dans la partie [Installation d' utilitaires] " 10 80 1

{ if
(whiptail --title "pihole" --yes-button "Non" --no-button "Oui" --yesno "Installer pihole - Un serveur dns pour bloquer les publicités ?" 10 80)
then
echo "$CURRENTUSER n' a pas activé pihole" | sudo tee --append /etc/startpi/log.txt
else
echo "$CURRENTUSER a activé pihole" | sudo tee --append /etc/startpi/log.txt

echo "Installation..."

curl -sSL https://install.pi-hole.net | bash

echo "Installation effectuée !"

whiptail --msgbox "Accédez à l' interfaçe pi-hole sur http://ip-de-votre-rpi/admin  " 10 80 1

    { if
    (whiptail --title "piholedns" --yes-button "Non" --no-button "Oui" --yesno "Installer Pi-hole comme un serveur DNS (ne redirigera pas les requêtes vers un DNS public - Meilleure vie privée) ?" 10 80)
    then
    echo "$CURRENTUSER n' a pas activé piholedns" | sudo tee --append /etc/startpi/log.txt
    else
    echo "$CURRENTUSER a activé piholedns" | sudo tee --append /etc/startpi/log.txt

    echo "Installation..."

    sudo apt install unbound
    wget https://www.internic.net/domain/named.root -O /var/lib/unbound/root.hints
    sudo service unbound start
    echo "Installation effectuée !"
    whiptail --msgbox "Vous devriez aller sur https://docs.pi-hole.net/guides/unbound/ pour vous renseigner davantage  " 10 80 1


    fi }

{ if
    (whiptail --title "piholedns" --yes-button "Non" --no-button "Oui" --yesno "Changer le mot de passe de pi-hole pour plus de sécurité ?" 10 80)
    then
    echo "$CURRENTUSER n' a pas activé piholedns" | sudo tee --append /etc/startpi/log.txt
    else
    echo "$CURRENTUSER a activé piholedns" | sudo tee --append /etc/startpi/log.txt

    echo "Chargement..."

    NEW_PASS=$(whiptail --inputbox "Entrez le mot de passe" 10 80 "" 3>&1 1>&2 2>&3)
pihole -a -p $NEW_PASS
    echo "Fait !"


    fi }

fi }


{ if
(whiptail --title "pyload" --yes-button "Non" --no-button "Oui" --yesno "Installer pyload - Un serveur de téléchargement ?" 10 80)
then
echo "$CURRENTUSER n' a pas activé pyload" | sudo tee --append /etc/startpi/log.txt
else
echo "$CURRENTUSER a activé pyload" | sudo tee --append /etc/startpi/log.txt

echo "Installation..."

sudo apt-get install python-support python-crypto python-pycurl tesseract-ocr tesseract-ocr-eng python-imaging python-pip python-dev pyopenssl rhino -y
sudo apt-get install libmozjs-24-bin
cd /usr/bin
ln -s js24 js
cp /etc/apt/sources.list /etc/startpi/backups/sources.list
sed -i "$ a deb http://mirrordirector.raspbian.org/raspbian/ jessie main contrib non-free rpi" /etc/apt/sources.list
sed -i "$ a deb http://mirrordirector.raspbian.org/raspbian/ jessie main contrib non-free rpi" /etc/apt/sources.list
sudo apt-get update
sudo apt-get -y install git liblept4 python python-crypto python-pycurl python-imaging tesseract-ocr zip unzip python-openssl libmozjs-24-bin
wget http://ftp.fr.debian.org/debian/pool/main/g/gcc-6/gcc-6-base_6.3.0-18+deb9u1_armhf.deb
dpkg -i gcc-6-base_6.3.0-18+deb9u1_armhf.deb
rm gcc-6-base_6.3.0-18+deb9u1_armhf.deb
wget http://ftp.fr.debian.org/debian/pool/main/g/gcc-6/libgcc1_6.3.0-18+deb9u1_armhf.deb
dpkg -i libgcc1_6.3.0-18+deb9u1_armhf.deb
rm libgcc1_6.3.0-18+deb9u1_armhf.deb
wget http://ftp.fr.debian.org/debian/pool/main/g/gcc-6/libstdc++6_6.3.0-18+deb9u1_armhf.deb
dpkg -i libstdc++6_6.3.0-18+deb9u1_armhf.deb
rm libstdc++6_6.3.0-18+deb9u1_armhf.deb
wget http://ftp.fr.debian.org/debian/pool/main/g/glibc/libc6_2.24-11+deb9u3_armhf.deb
dpkg -i libc6_2.24-11+deb9u3_armhf.deb
rm libc6_2.24-11+deb9u3_armhf.deb
wget http://ftp.fr.debian.org/debian/pool/non-free/u/unrar-nonfree/unrar_5.3.2-1+deb9u1_armhf.deb
dpkg -i unrar_5.3.2-1+deb9u1_armhf.deb
rm unrar_5.3.2-1+deb9u1_armhf.deb
sudo apt-get -y build-dep rar unrar-nonfree
sudo apt-get source -b unrar-nonfree
sudo dpkg -i unrar_*_armhf.deb
sudo rm -rf unrar-*
sudo adduser --system pyload
cd /opt
sudo git clone -b stable https://github.com/pyload/pyload.git
cd pyload
whiptail --msgbox "Vous allez ici configurer pyload  " 10 80 1
sudo -u pyload python pyLoadCore.py
sudo echo """[Unit]
Description=Python Downloader
After=network.target

[Service]
User=pyload
ExecStart=/usr/bin/python /opt/pyload/pyLoadCore.py

[Install]
WantedBy=multi-user.target""" > /etc/systemd/system/pyload.service
sudo systemctl enable pyload.service
sudo service pyload restart
echo "Installation effectuée !"


fi }

################################################################
##                   Stopped French Version.                  ##
################################################################
